package main;

import java.util.*;
import java.util.stream.Collectors;

public class UseCase8TrainConsisntMgmt {

    // Make this PUBLIC STATIC (important!)
    public static class Bogie {
        private String type;
        private int capacity;

        public Bogie(String type, int capacity) {
            this.type = type;
            this.capacity = capacity;
        }

        public int getCapacity() {

            return capacity;
        }
    }

    public static List<Bogie> filterBogies(List<Bogie> bogies, int threshold) {
        return bogies.stream()
                .filter(b -> b.getCapacity() > threshold)
                .collect(Collectors.toList());
    }
}